### PR TITLE
DOCU-1143 changes

### PR DESCRIPTION
**Pull request by lucasdigital**
HVM-1205 introduces new functionality for viewing and changing a LDM's truststore.

These changes are added to the configuration-ui page because this is where we have covered SSL/TLS. A best-fit, not a perfect fit.

I maintained the same formatting for presenting CURL commands that we already use - see Migration Gates section.

Note that we have previously referred to truststore as TrustStore. I think this was a Fusion docs approach but it isn't supported by a quick check of general use. I've therefore updated existing references to "truststore" and "Truststore" in titles.

As part of testing I discovered that the API documentation can only be accessed from the host machine itself. While this is an understandable approach from a security point-of-view, we need to clarify this in the API ref section and explain how to open access.

### CODE REVIEW

File: docs/configuration-ui.md [hunk 1]
```diff
@@ -222,6 +222,29 @@ keytool -import -trustcacerts -alias wandisco-ui -file server_cert.key -keystore
 See the [Oracle documentation](https://docs.oracle.com/cd/E19906-01/820-4916/geygn/index.html) for details on the parameters used.
 :::
 
+### Configure Truststore
```
Comment (AshVic) [inline docs/configuration-ui.md:225]: Configure a (or the) truststore (sentence style)

File: docs/configuration-ui.md [hunk 2]
```diff
@@ -222,6 +222,29 @@ keytool -import -trustcacerts -alias wandisco-ui -file server_cert.key -keystore
...
+
+The truststore used by LiveData Migrator can be viewed and updated through the [REST API](./api-reference/).
+
+#### View Truststore
```
Comment (AshVic) [inline docs/configuration-ui.md:229]: View a (or the) truststore

File: docs/configuration-ui.md [hunk 3]
```diff
@@ -222,6 +222,29 @@ keytool -import -trustcacerts -alias wandisco-ui -file server_cert.key -keystore
...
+
+View the existing truststore parameters with a GET query sent to `/config/ldm`.
+
+  For example:
+
+  ```text title='Example'
+  curl http://localhost:18080/config/ldm/
+  ```
+#### Configure Truststore
```
Comment (AshVic) [inline docs/configuration-ui.md:238]: Why do we repeat "configure the truststore" as a subheading of "Configure the truststore" which is the page title?

Comment (AshVic) [reply docs/configuration-ui.md:238]: Reading text below suggests it should perhaps be "Update the truststore" (update being the term we use for editing or changing)?

File: docs/configuration-ui.md [hunk 4]
```diff
@@ -222,6 +222,29 @@ keytool -import -trustcacerts -alias wandisco-ui -file server_cert.key -keystore
 See the [Oracle documentation](https://docs.oracle.com/cd/E19906-01/820-4916/geygn/index.html) for details on the parameters used.
 :::
 
+### Configure Truststore
+
+The truststore used by LiveData Migrator can be viewed and updated through the [REST API](./api-reference/).
```
Comment (AshVic) [inline docs/configuration-ui.md:227]: You can view and update...(use active voice).

File: docs/configuration-ui.md [hunk 5]
```diff
@@ -143,11 +143,11 @@ Once you've finished making changes to group privileges, click **Apply** to save
...
 ### Using LDAPS
 
-To use Secure LDAP (LDAPS), a trusted certificate must be present in the UI JVM TrustStore matching the certificate presented by the LDAP server.
-By default, the JVM running the UI uses a TrustStore called cacerts. 
+To use Secure LDAP (LDAPS), a trusted certificate must be present in the UI JVM truststore matching the certificate presented by the LDAP server.
+By default, the JVM running the UI uses a truststore called cacerts.
 
 To use a self-issued LDAP server certificate, you must import the certificate for this private certificate authority into the cacerts file.
-See the [Oracle documentation](https://docs.oracle.com/javase/tutorial/security/toolsign/rstep2.html) for steps to import a trusted certificate into a TrustStore.
+See the [Oracle documentation](https://docs.oracle.com/javase/tutorial/security/toolsign/rstep2.html) for steps to import a trusted certificate into a truststore.
```
Comment (AshVic) [inline docs/configuration-ui.md:150]: For more information about steps to import a trusted certificate to a truststore, see the Oracle documentation...(i think this is our standard phrasing).

File: docs/api-reference.md [hunk 6]
```diff
@@ -17,6 +17,24 @@ This version will reflect exactly the API exposed from your instance of LiveData
...
+LiveData Migrator's API documentation can only be viewed from the LDM host machine. This approach minimises the risk of unauthorized access. However, if you need to access the documentation over your network, take the following steps:
+
+1. Open `/etc/wandisco/livedata-migrator/application.properties`.
+1. Delete or comment out the `server.address=127.0.0.0` line.
+1. Save the change.
+1. Restart the LiveData Migrator service to apply the change:
+  ```
+  restart livedata-migrator.service
+  ```
+1. Check that access to the documentation is now available on your network:
```
Comment (AshVic) [inline docs/api-reference.md:32]: Check you can access the documentation on your network by running the command: xyz.
We drop "that" and address user in first person with active voice.

File: docs/api-reference.md [hunk 7]
```diff
@@ -17,6 +17,24 @@ This version will reflect exactly the API exposed from your instance of LiveData
 
 [LiveData Migrator Swagger Documentation](/img/swagger.png)
 
+
+### Access REST API documentation on your network
+
+LiveData Migrator's API documentation can only be viewed from the LDM host machine. This approach minimises the risk of unauthorized access. However, if you need to access the documentation over your network, take the following steps:
```
Comment (AshVic) [inline docs/api-reference.md:23]: You can view...(active voice, address user in first person).

File: docs/api-reference.md [hunk 8]
```diff
@@ -17,6 +17,24 @@ This version will reflect exactly the API exposed from your instance of LiveData
...
+
+1. Open `/etc/wandisco/livedata-migrator/application.properties`.
+1. Delete or comment out the `server.address=127.0.0.0` line.
+1. Save the change.
```
Comment (AshVic) [inline docs/api-reference.md:27]: How does the user save the change?

File: docs/configuration-ui.md [hunk 9]
```diff
@@ -143,11 +143,11 @@ Once you've finished making changes to group privileges, click **Apply** to save
 
 ### Using LDAPS
 
-To use Secure LDAP (LDAPS), a trusted certificate must be present in the UI JVM TrustStore matching the certificate presented by the LDAP server.
-By default, the JVM running the UI uses a TrustStore called cacerts. 
+To use Secure LDAP (LDAPS), a trusted certificate must be present in the UI JVM truststore matching the certificate presented by the LDAP server.
```
Comment (AshVic) [inline docs/configuration-ui.md:146]: To use LDAPS (you already used the abbreviation so you don't need to write in full again), you need a trusted certificate in the UI JVM truststore that matches the certificate presented by the LDAP server.

File: docs/configuration-ui.md [hunk 10]
```diff
@@ -143,11 +143,11 @@ Once you've finished making changes to group privileges, click **Apply** to save
...
+By default, the JVM running the UI uses a truststore called cacerts.
```
Comment (AshVic) [inline docs/configuration-ui.md:147]: Which font do we use for calling out a technical name? In this case: cacerts.

File: docs/configuration-ui.md [hunk 11]
```diff
@@ -210,10 +210,10 @@ Configure how the UI uses TLS, which is disabled by default.
 | `server.ssl.key-store-password` | The password to be used to access the key store |
 | `server.ssl.key-alias` | The alias of the certificate to be used |
 | `server.ssl.key-store-type` | Optional: set the key store type. Defaults to `PKCS12` |
-| `application.liveMigratorV2.client.noCheckCertificate` | Optional: add this property and set the value to `true` if you want to implicitly trust certificates from remote LiveData Migrator instances. <br/>**Default value**: `false` <br/><br/> It is not recommended to use this property unless you are fully aware of the consequences. A safer method is to import your server certificate inside a TrustStore (see the tip below for an example). |
+| `application.liveMigratorV2.client.noCheckCertificate` | Optional: add this property and set the value to `true` if you want to implicitly trust certificates from remote LiveData Migrator instances. <br/>**Default value**: `false` <br/><br/> It is not recommended to use this property unless you are fully aware of the consequences. A safer method is to import your server certificate inside a truststore (see the tip below for an example). |
```
Comment (AshVic) [inline docs/configuration-ui.md:213]: We don't recommend using this property unless you're...(I would rephrase the bit about being "aware of the consequences" in a slightly less threatening way?)...Can you just say:
"We recommend you import your server certificate inside a truststore."?
Drop the first sentence and the additional  (see the tip below for an example). It's already called out by the tip paragraph.

File: docs/configuration-ui.md (hunk 10 above)
Comment (AshVic) [reply docs/configuration-ui.md:147]: Ah i see below you have `cacerts` - use the same for consistency.

File: docs/configuration-ui.md [hunk 12]
```diff
@@ -222,6 +222,29 @@ keytool -import -trustcacerts -alias wandisco-ui -file server_cert.key -keystore
 See the [Oracle documentation](https://docs.oracle.com/cd/E19906-01/820-4916/geygn/index.html) for details on the parameters used.
```
Comment (AshVic) [inline docs/configuration-ui.md:222]: For more information about parameters, see...

**Review by AshVic — CHANGES_REQUESTED** [review]

File: docs/configuration-ui.md (hunk 3 above)
Comment (lucasdigital) [reply docs/configuration-ui.md:238]: Exactly so! I think I missed making this suggested update. Now done!

File: docs/configuration-ui.md (hunk 10 above)
Comment (lucasdigital) [reply docs/configuration-ui.md:147]: I've updated that instance of cacerts to use the code formatting `cacerts`

File: docs/configuration-ui.md (hunk 12 above)
Comment (lucasdigital) [reply docs/configuration-ui.md:222]: Updated!

Comment (lucasdigital) [reply docs/configuration-ui.md]: All changes now in place.
